From 010413e2f08cfa08a36d1937683ec7e8d4f9acdb Mon Sep 17 00:00:00 2001
From: tanzby <tanzby@outlook.com>
Date: Wed, 29 Mar 2023 22:05:28 +0800
Subject: [PATCH] patch

---
 BUILD.bazel | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/BUILD.bazel b/BUILD.bazel
index 7aecb3e0..3881d684 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -36,7 +36,7 @@ cc_library(
         "include/oneapi/tbb/detail/*.h",
     ]),
     copts = ["-w"] + select({
-        "@bazel_tools//platforms:windows": [""],
+        "@platforms//os:windows": [""],
         "//conditions:default": ["-mwaitpkg"],
     }),
     defines =
@@ -47,7 +47,7 @@ cc_library(
             ],
         }) +
         select({
-            "@bazel_tools//platforms:osx": ["_XOPEN_SOURCE"],
+            "@platforms//os:osx": ["_XOPEN_SOURCE"],
             "//conditions:default": [],
         }),
     includes = [
@@ -55,8 +55,8 @@ cc_library(
     ],
     linkopts =
         select({
-            "@bazel_tools//platforms:windows": [],
-            "@bazel_tools//platforms:linux": [
+            "@platforms//os:windows": [],
+            "@platforms//os:linux": [
                 "-ldl",
                 "-pthread",
                 "-lrt",
-- 
2.37.1 (Apple Git-137.1)

